src/wallet/wallet.h

```diff
@@ -665,6 +720,346 @@ class CWallet : public CCryptoKeyStore, public CValidationInterface
      */
     bool SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAmount& nTargetValue, std::set<std::pair<const CWalletTx*,unsigned int> >& setCoinsRet, CAmount& nValueRet, const CCoinControl *coinControl = NULL, bool fForUseInInstantSend = true) const;
 
+    void AddToSpends(const COutPoint &outpoint, const uint256 &wtxid);
+
+    void AddToSpends(const uint256 &wtxid);
+
+    CAmount GetFee(const CCoinControl* coinControl, size_t txSize);
+    std::vector<CTransparentTxout> GetTransparentTxouts() const;
+
+    // Get all the vRelevantTransactions that can be used. Inputs specifically selected in coinControl are put into
+    // vCoinControlInputs, and all other inputs compatible with coinControl and available for use are put into
+    // vAvailableInputs.
+    template<typename AbstractTxout>
+    void GetAvailableInputs(const std::vector<AbstractTxout>& vRelevantTransactions,
```

Comment:
This is a huge function to leave it in the header file. Please consider moving it to cpp file and then explicitly instantiate it with all the possible AbstractTxout types